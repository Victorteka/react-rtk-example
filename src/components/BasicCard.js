import * as React from 'react';
import { Link } from 'react-router-dom';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';


export default function BasicCard({ readMoreClicked, post }) {

  const readMore = !readMoreClicked

  return (
    <Card sx={{ minWidth: 275 }} style={{marginTop:8}} data-testid='basic-card'>
      <CardContent>
        <Typography variant='h6'  sx={{ fontSize: 16 }} gutterBottom>
          {post.title}
        </Typography>
        <Typography variant="body2">
          {readMore ? post.body : post.body.substring(0,100)} 
        </Typography>
      </CardContent>
      <CardActions>
        {
          readMore ? null : <Link to={`/posts/${post.id}`}><Button size="small">Read More</Button></Link>
        }
      </CardActions>
    </Card>
  );
}