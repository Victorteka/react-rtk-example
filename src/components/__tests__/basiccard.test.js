import { cleanup, render, screen } from "@testing-library/react"
import renderer from 'react-test-renderer'
import BasicCard from '../BasicCard'

afterEach(() => {
    cleanup()
})

test('should render BasicCard component', () => {
    const post = {title: 'title', body:'body'}
    render(<BasicCard post={post} />)
    const basicCard = screen.getByTestId('basic-card')
    expect(basicCard).toBeInTheDocument()
    expect(basicCard).toHaveTextContent("title")
})

test('matches snapshot', () => {
    const post = {title: 'title', body:'body'}
    const tree = renderer.create(<BasicCard post={post} />).toJSON()
    expect(tree).toMatchSnapshot()
})