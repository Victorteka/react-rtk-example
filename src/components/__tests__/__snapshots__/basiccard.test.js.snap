// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
  data-testid="basic-card"
  style={
    Object {
      "marginTop": 8,
    }
  }
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-7hk7nn-MuiTypography-root"
    >
      title
    </h6>
    <p
      className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
    >
      body
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  />
</div>
`;
